packages/features/ee/managed-event-types/lib/handleChildrenEventTypes.ts

```diff
@@ -255,7 +240,12 @@ export default async function handleChildrenEventTypes({
           },
           data: {
             ...updatePayloadFiltered,
-            hashedLink: "hashedLink" in unlockedFieldProps ? undefined : hashedLinkQuery(userId),
+            hashedLink:
+              "singleUseLinks" in unlockedFieldProps
+                ? undefined
+                : {
+                    deleteMany: {},
+                  },
```

Comment:
In case admin does not allow hashed links, delete all the hashedlinks which might be present previously